src/hooks/useMaxAmountIn.test.tsx

```diff
@@ -0,0 +1,16 @@
+import { TEST_ALLOWED_SLIPPAGE, TEST_TRADE_EXACT_INPUT } from 'test-utils/constants'
+import { renderHook } from 'test-utils/render'
+
+import { useMaxAmountIn } from './useMaxAmountIn'
+
+describe('useMaxAmountIn', () => {
+  it('returns undefined for an undefined trade', () => {
+    const { result } = renderHook(() => useMaxAmountIn(undefined, TEST_ALLOWED_SLIPPAGE))
+    expect(result.current).toEqual(undefined)
+  })
+
+  it('returns the expected value calculated using the exact-input trade and the given slippage', () => {
+    const { result } = renderHook(() => useMaxAmountIn(TEST_TRADE_EXACT_INPUT, TEST_ALLOWED_SLIPPAGE))
+    expect(result.current?.toExact()).toEqual('0.000000000000001')
```

Comment:
It's not obvious what TEST_TRADE_EXACT_INPUT is from looking at this file. Could you derive the expected result from that const, or else use a literal in L13?